engine/src/main/java/org/hibernate/search/engine/metadata/impl/AnnotationMetadataProvider.java

```diff
@@ -385,6 +385,11 @@ private void createIdPropertyMetadata(XProperty member,
 
 		Field.TermVector termVector = AnnotationProcessingHelper.getTermVector( TermVector.NO );
 
+		if ( !parseContext.skipAnalyzers() ) {
+			// Register a pass through analyzer for the document id
+			typeMetadataBuilder.addToScopedAnalyzerReference( fieldPath, null, Field.Index.NOT_ANALYZED );
+		}
```

Comment:
I think you also need to fix `createPropertyMetadataForEmbeddedId`. It's the one called when including the ID in an `@IndexedEmbedded`, either though `@IndexedEmbedded.includePaths` or `@IndexedEmbedded.includeEmbeddedObjectId`.
And we need a test for it, too.

Reply:
Added a commit to take this case into account and rebased.